components/nifosg/nifloader.cpp

```diff
@@ -1289,50 +1280,81 @@ namespace NifOsg
                         std::string filename = Misc::ResourceHelpers::correctTexturePath(st->filename, textureManager->getVFS());
 
                         unsigned int clamp = static_cast<unsigned int>(tex.clamp);
-                        int wrapT = (clamp) & 0x1;
-                        int wrapS = (clamp >> 1) & 0x1;
+                        bool wrapT = (clamp) & 0x1;
+                        bool wrapS = (clamp >> 1) & 0x1;
 
-                        osg::Texture2D* texture2d = textureManager->getTexture2D(filename,
+                        osg::ref_ptr<osg::Texture2D> texture2d = textureManager->getTexture2D(filename,
                               wrapS ? osg::Texture::REPEAT : osg::Texture::CLAMP,
                               wrapT ? osg::Texture::REPEAT : osg::Texture::CLAMP);
 
                         int texUnit = boundTextures.size();
 
-                        stateset->setTextureAttributeAndModes(texUnit, texture2d, osg::StateAttribute::ON);
-
-                        if (i == Nif::NiTexturingProperty::GlowTexture)
+                        osg::ref_ptr<osg::TexEnv> texEnv = NULL;
+                        switch(i)
                         {
-                            osg::TexEnvCombine* texEnv = new osg::TexEnvCombine;
-                            texEnv->setCombine_Alpha(osg::TexEnvCombine::REPLACE);
-                            texEnv->setSource0_Alpha(osg::TexEnvCombine::PREVIOUS);
-                            texEnv->setCombine_RGB(osg::TexEnvCombine::ADD);
-                            texEnv->setSource0_RGB(osg::TexEnvCombine::PREVIOUS);
-                            texEnv->setSource1_RGB(osg::TexEnvCombine::TEXTURE);
-
-                            stateset->setTextureAttributeAndModes(texUnit, texEnv, osg::StateAttribute::ON);
+                        case Nif::NiTexturingProperty::BaseTexture:
+                            texEnv = new osg::TexEnv;
```

Comment:
Just pointing out for completion sake that this line was redundant and would have caused additional API overhead as well.